examples/music-v2.py

```diff
@@ -260,6 +282,29 @@ def __init__(self, bot):
         await player.disconnect()
         await ctx.send('*⃣ | Disconnected.')
 
+    @_join.before_invoke
+    async def ensure_voice(self, ctx):
+        """ A few checks to make sure the bot can join a voice channel. """
+        player = self.bot.lavalink.players.get(ctx.guild.id)
+
+        if not ctx.author.voice or not ctx.author.voice.channel:
+            await ctx.send("You aren't connected to any voice channel.")
+            raise commands.CommandInvokeError("Author not connected to voice channel.")
+
+        if not player.is_connected:
+            permissions = ctx.author.voice.channel.permissions_for(ctx.me)
+
+            if not permissions.connect or not permissions.speak:
+                await ctx.send('Missing permissions `CONNECT` and/or `SPEAK`.')
+                raise commands.CommandInvokeError("Bot has no permissions CONNECT and/or SPEAK")
+
+
+bot = commands.Bot(command_prefix='lava.', description="A simple music bot that uses Lavalink.", case_insensitive=True)
+bot.add_cog(Music(bot))
+
+
+@bot.event

```

Comment:
Yes, I know. But I personally know enough people who don't use the commands extension. Instead, they write their own command parsers. Lavalink forces the users to use commands.Bot because of the `on_socket_response` event. So I thought it would be better for people who are new to the commands extension to show them how a simple bot could look like.

Reply:
I still believe this should be kept as a cog more than anything else — to serve as a reference. I know a lot of people that use the example as a cog (myself included). Additionally, I strongly believe that someone new to Python/discord.py shouldn't try to learn the language/library with a music bot, there are much simpler "getting started" examples out there.